File: redsheet-portal/src/app/database-models/v-project-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vProjectRecord {
    constructor(vProjectRecord?: vProjectRecord )  {
        let properties = [ "isClosed", "ownerId", "ownerName", "projectId", "projectName", "projectNote", "status", "templateGroupId", "templateGroupName", "templateId",];
        for (let property of properties)
            if (vProjectRecord && vProjectRecord[property])
                this[property] = vProjectRecord[property];
    }
 
	isClosed : boolean = false ;
 
    @range(1,2147483647)
	ownerId : number =   undefined;
 
    @required()
    @maxLength(201)
	ownerName : string =   undefined;
 
	projectId : number =   0 ;
 
    @required()
    @maxLength(200)
	projectName : string =   undefined;
 
    @maxLength(1000)
	projectNote : string =   undefined;
 
	status : boolean = false ;
 
    @required()
	templateGroupId : string =   undefined;
 
    @required()
    @maxLength(500)
	templateGroupName : string =   undefined;
 
    @range(1,2147483647)
	templateId : number =   undefined;


}

File: RedSheet.DbEntities/Models/vProjectRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vProjectRecords", Schema = "dbo")]
	[Serializable]
	public partial class vProjectRecord  
	{
 
	[Required]
	public bool IsClosed { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int OwnerId { get; set; } 
 
	[Required] 
	[MaxLength(201)]
	public string OwnerName { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int ProjectId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string ProjectName { get; set; } 
 
	[MaxLength(1000)]
	public string ProjectNote { get; set; } 
 
	[Required]
	public bool Status { get; set; } 
 
	[Required]
	public Guid TemplateGroupId { get; set; } 
 
	[Required] 
	[MaxLength(500)]
	public string TemplateGroupName { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int TemplateId { get; set; } 

	public vProjectRecord()
	{
    }

    }
}

